### PR TITLE
hapi 17

### PR DESCRIPTION
**Pull request by kdong007**
https://github.com/apollographql/apollo-engine-js/issues/85

### CODE REVIEW

**Comment by kdong007** [comment]
first time doing pull request. let me know if anything goes wrong

**Comment by zionts** [comment]
Hey @kdong007 thanks so much for opening up a PR to upgrade to Hapi 17! Unfortunately, it looks like tests for Hapi are failing with this patch. I did some work on this earlier, and I found that the [error](https://github.com/apollographql/apollo-engine-js/blob/adam/upgradehapi/src/middleware.ts#L83) is probably somewhere in the `proxyRequest` step. I encourage you to check out [my branch](https://github.com/apollographql/apollo-engine-js/tree/adam/upgradehapi) and work off of it (after a rebase, probably). The important thing is that the tests are passing. To check that, from the root of the directory, just run `npm test` and if all tests pass, that's your first bar 👍 
I'm happy to help you out if you get stuck, too! If you think you've made some progress beyond what I have on the branch I mentioned, feel free to open up a PR (even if the tests are failing) and let me know where you're stuck so I can help you dig into it.

File: src/middleware.ts [hunk 1]
```diff
@@ -74,12 +74,13 @@ export function makeKoaMiddleware(params: MiddlewareParams) {
...
-        if (!params.uri) return reply.continue();
+        if (!params.uri) return reply.continue;
         const path = req.url.pathname;
-        if (!path || path !== params.endpoint) return reply.continue();
-        else if (req.method !== 'get' && req.method !== 'post') return reply.continue();
-        else if (req.headers['x-engine-from'] === params.psk) return reply.continue();
+        if (!path || path !== params.endpoint) return reply.continue;
+        else if (req.method !== 'get' && req.method !== 'post') return reply.continue;
+        else if (req.headers['x-engine-from'] === params.psk) return reply.continue;
         else proxyRequest(params, req.raw.req, req.raw.res);
```
Comment (zionts) [inline src/middleware.ts]: Something goes awry in this step – there also need to be changes made to the tests, but I think I made most of them on the branch I pointed to in my comment :)

**Comment by zionts** [comment]
@kdong007 I'm going to close this one up and point you at https://github.com/apollographql/apollo-engine-js/pull/116 – I'm not very familiar with Hapi, but I got all tests without the proxying step to work. If you can see where it's going awry, please let me know! Thanks for putting this together and please feel free to continue contributing on the PR I pointed to above :)

**Pull request closed without merging**